sqlmesh/core/engine_adapter/redshift.py

```diff
@@ -125,10 +125,18 @@ def _build_create_table_exp(
             # is applied to a ctas statement, VARCHAR types default to 1 in some instances.
             # this checks the explain plain from redshift and tries to detect when these optimizer
             # bugs occur and force a cast
-            sql = statement.sql(
+            explain_statement = statement.copy()
+            for select in explain_statement.find_all(exp.Select):
+                if select.args.get("from"):
+                    select.set("limit", None)
+                    select.set("where", None)
+
+            explain_statement_sql = explain_statement.sql(
```

Comment:
copy=False here so that there isn't a double copy, you explicitly copy on 128 now.  sql() calls always perform a copy by default because some sql calls do mutations. but in this case you don't care, so we don't need the extra overhead